build-scripts/createRelease.ts

```diff
@@ -0,0 +1,62 @@
+/*!
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as child_process from 'child_process'
+import * as fs from 'fs-extra'
+import * as path from 'path'
+
+const changesDirectory = '.changes'
+const nextReleaseDirectory = path.join(changesDirectory, 'next-release')
+// tslint:disable-next-line:no-var-requires no-unsafe-any
+const releaseVersion = require(path.join('..', 'package.json')).version
+const changesFile = path.join(changesDirectory, `${releaseVersion}.json`)
+
+fs.mkdirpSync(nextReleaseDirectory)
+
+const changeFiles = fs.readdirSync(nextReleaseDirectory)
+if (changeFiles.length === 0) {
+    console.log('Error! no changes to release!')
+    process.exit(-1)
```

Comment:
Would this cause a release pipeline to fail?
I'm guessing a situation where we had a release containing implementation changes only could result in a release without any changelog entries, in which case we would go an add one saying something along the lines of (for example) "this release contains performance improvements under the hood", and re-run the pipeline.

Reply:
This would fail, which is how our other pipelines work. I think it makes sense because it forces you to at least write that to the changelog.